modules/phase_field/src/kernels/NonlinearDiffusion.C

```diff
@@ -0,0 +1,25 @@
+#include "NonlinearDiffusion.h"
+
+registerMooseObject("PhaseFieldApp", NonlinearDiffusion);
+
+InputParameters
+NonlinearDiffusion::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addClassDescription("Compute the nonlinear diffusion term");
+  params.addRequiredParam<std::string>("D", "diffusivity for the diffusion equation");
```

Comment:
You should be using `MaterialPropertyName` instead of `std::string` for this everywhere